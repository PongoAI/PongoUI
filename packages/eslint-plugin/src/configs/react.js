/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
