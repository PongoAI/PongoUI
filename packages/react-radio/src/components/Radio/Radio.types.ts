import type { RadioProps as FluentRadioProps, RadioState as FluentRadioState } from '@fluentui/react-radio';

export type RadioProps = FluentRadioProps;

export type RadioState = FluentRadioState;
