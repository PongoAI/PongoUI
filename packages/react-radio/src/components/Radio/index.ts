export * from './Radio';
export * from './useRadioStyles';
export * from './Radio.types';
