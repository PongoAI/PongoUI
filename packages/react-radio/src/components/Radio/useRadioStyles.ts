import { mergeClasses, makeStyles, shorthands } from '@griffel/react';
import { tokens } from '@pongo-ui/react-theme';
import type { RadioState } from './Radio.types';
import { createCustomFocusIndicatorStyle } from '@fluentui/react-tabster';

const indicatorClassName = 'pongo-radio-indicator';
const labelClassName = 'pongo-radio-label';

const useRootStyles = makeStyles({
  root: {
    display: 'inline-flex',
    position: 'relative',
    columnGap: '8px',
    alignItems: 'center',
  },

  vertical: {
    flexDirection: 'column',
    alignItems: 'center',
    rowGap: '8px',
  },

  focusIndicator: createCustomFocusIndicatorStyle(
    {
      ':after': {
        content: "''",
        position: 'absolute',
        top: '-4px',
        right: '-4px',
        bottom: '-4px',
        left: '-4px',
        boxSizing: 'border-box',
        ...shorthands.border('2px', 'solid', tokens.textColor),
        ...shorthands.borderRadius(tokens.square),
      },
    },
    { selector: 'focus-within' },
  ),
});

const useInputStyles = makeStyles({
  input: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: '100%',
    height: '100%',
    boxSizing: 'border-box',
    opacity: 0,
    ...shorthands.margin(0),

    [`:not(:checked) ~ .${indicatorClassName} > *`]: {
      transform: 'scale(0%)',
    },
  },

  enabled: {
    cursor: 'pointer',
    ':enabled:not(:checked)': {
      [`& ~ .${labelClassName}`]: {
        color: tokens.inherit,
      },
      [`& ~ .${indicatorClassName}`]: {
        ...shorthands.borderColor(tokens.brand),
      },

      ':hover': {
        [`& ~ .${labelClassName}`]: {
          color: tokens.inheritHover,
        },
        [`& ~ .${indicatorClassName}`]: {
          ...shorthands.borderColor(tokens.brandHover),
        },
      },

      ':hover:active': {
        [`& ~ .${labelClassName}`]: {
          color: tokens.inheritPressed,
        },
        [`& ~ .${indicatorClassName}`]: {
          ...shorthands.borderColor(tokens.brandPressed),
        },
      },
    },

    ':enabled:checked': {
      [`& ~ .${labelClassName}`]: {
        color: tokens.textColor,
      },
      [`& ~ .${indicatorClassName}`]: {
        ...shorthands.borderColor(tokens.brand),
        color: tokens.brand,
      },

      ':hover': {
        [`& ~ .${indicatorClassName}`]: {
          ...shorthands.borderColor(tokens.brandHover),
          color: tokens.brandHover,
        },
      },

      ':hover:active': {
        [`& ~ .${indicatorClassName}`]: {
          ...shorthands.borderColor(tokens.brandPressed),
          color: tokens.brandPressed,
        },
      },
    },
  },

  disabled: {
    cursor: 'not-allowed',
    [`& ~ .${labelClassName}`]: {
      color: tokens.inheritDisabled,
    },
    [`& ~ .${indicatorClassName}`]: {
      ...shorthands.borderColor(tokens.inheritDisabled),
      color: tokens.inheritDisabled,
    },
  },
});

const useIndicatorStyles = makeStyles({
  indicator: {
    display: 'flex',
    width: '16px',
    height: '16px',
    boxSizing: 'border-box',
    flexShrink: 0,
    alignItems: 'center',
    justifyContent: 'center',
    pointerEvents: 'none',

    ...shorthands.overflow('hidden'),
    ...shorthands.border('2px', 'solid'),
    ...shorthands.borderRadius(tokens.circle),

    '& svg': {
      width: '10px',
      fill: tokens.brand,
      transitionProperty: ' transform',
      transitionDuration: '.2s',
      transitionTimingFunction: 'cubic-bezier(0.4, 0, 0.2, 1) 0m',
    },
  },
});

const useLabelStyles = makeStyles({
  label: {
    alignSelf: 'center',
    userSelect: 'none',
    fontFamily: tokens.fontFamilyBase,
    color: tokens.textColor,
    fontSize: tokens.fontSize300,
  },
});

export const useRadioStyles = (state: RadioState): RadioState => {
  const rootStyles = useRootStyles();
  const labelStyles = useLabelStyles();
  const indicatorStyles = useIndicatorStyles();
  const inputStyles = useInputStyles();

  state.root.className = mergeClasses(
    rootStyles.root,
    rootStyles.focusIndicator,
    state.labelPosition === 'below' && rootStyles.vertical,
    state.root.className,
  );

  state.input.className = mergeClasses(
    inputStyles.input,
    state.input.disabled ? inputStyles.disabled : inputStyles.enabled,
    state.input.className,
  );

  if (state.label) {
    state.label.className = mergeClasses(labelClassName, labelStyles.label, state.label.className);
  }

  if (state.indicator) {
    state.indicator.className = mergeClasses(indicatorClassName, indicatorStyles.indicator, state.indicator.className);
  }

  return state;
};
