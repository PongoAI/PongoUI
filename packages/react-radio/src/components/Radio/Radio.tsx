import * as React from 'react';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import { renderRadio_unstable, useRadio_unstable } from '@fluentui/react-radio';
import type { RadioProps } from './Radio.types';
import { useRadioStyles } from './useRadioStyles';

/**
 * Radio component is a wrapper for a radio button with a label.
 */
export const Radio: ForwardRefComponent<RadioProps> = React.forwardRef((props, ref) => {
  const state = useRadio_unstable(props, ref);

  useRadioStyles(state);

  return renderRadio_unstable(state);
});
Radio.displayName = 'Radio';
