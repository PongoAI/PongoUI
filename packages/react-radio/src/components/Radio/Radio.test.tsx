import React from 'react';
import { Radio } from './Radio';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('Radio', () => {
  isConformant({
    Component: Radio,
    displayName: 'Radio',
  });

  it('renders correctly', () => {
    const result = render(<Radio>Test</Radio>);
    expect(result.container).toMatchSnapshot();
  });
});
