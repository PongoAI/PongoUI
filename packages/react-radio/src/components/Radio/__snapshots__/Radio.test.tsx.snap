// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<div>
  <span
    class="___18rr115 ftuwxu6 f10pi13n fnmxc1e f122n59 f1ewtqcl fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f8qjfyk f2hkw1w fdxsi5y f1i4q40k fz5pyj9 fuitljl f1lxa25c f1dexcea fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fgm4zzz f1o1s7a4 f1w999bl fr2bnm2 f1jjwxw3 fijdmwi f11pcu7g f1ddvjn"
  >
    <input
      class="___1k5a78d f1euv43f f1vgc2s3 f15twtuk f1e31b4d f1yab3r1 fly5x3f f1l02sjl f1ewtqcl fk73vx1 f1hu3pq6 f11qmguv f19f4twv f1tyq0we fowr46a f1xl0qwy f1k6fduh fkavele fg1nono fexw84i f1y6vlxy feznpzr f1ym48ab fxyywaq fc390eo f1julexz f1qlwis7 f13bd10 fiepdt6 f1fxl38v f1vmirnt f16aqadv fgoi767 f1qzxoeo f1rwhajm fxia3e6 fdxoyg9 ftb0ayp f1wkbyjn f7ombvi f49rsrr fpucs8 fsri65q f1ur2ra4 f1mu7ado f1o3leve f1s168r8 fdjrdmx"
      id="radio-8"
      type="radio"
      value="2"
    />
    <div
      aria-hidden="true"
      class="pongo-radio-indicator ___n55l230 f22iagw fjw5fx7 fd461yt f1ewtqcl fi64zpg f122n59 f4d9j23 f1aehjj5 f1p9o1ba f1sil6mw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f19qt5od ftovyd4 f1fajnbs ffz4t7b f1wac4k f1t8un4"
    >
      <svg
        aria-hidden="true"
        class="___1yjlmar f1w7gpdv fez10in f1kxrtqx"
        fill="currentColor"
        height="1em"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 2a8 8 0 100 16 8 8 0 000-16z"
        />
      </svg>
    </div>
    <label
      class="fui-Label pongo-radio-label ___1fy0nvb fk6fouc fc467gg f1arxnym f1i3iumi f7nlbp4 f1xqy1su"
      for="radio-8"
    >
      hello
    </label>
  </span>
</div>
`;
